diff --git a/configure.ac b/configure.ac
index acff772..1d9a985 100644
--- a/configure.ac
+++ b/configure.ac
@@ -36,7 +36,7 @@ AC_CONFIG_FILES([Makefile src/Makefile \
 
 LIBBLOCKDEV_PKG_CHECK_MODULES([GLIB], [glib-2.0 >= 2.42.2])
 LIBBLOCKDEV_PKG_CHECK_MODULES([GIO], [gio-2.0 >= 2.42.2])
-LIBBLOCKDEV_PKG_CHECK_MODULES([CRYPTSETUP], [libcryptsetup >= 1.6.7])
+LIBBLOCKDEV_PKG_CHECK_MODULES([CRYPTSETUP], [libcryptsetup >= 1.6.6])
 LIBBLOCKDEV_PKG_CHECK_MODULES([NSS], [nss >= 3.18.0])
 LIBBLOCKDEV_PKG_CHECK_MODULES([DEVMAPPER], [devmapper >= 1.02.93])
 LIBBLOCKDEV_PKG_CHECK_MODULES([UDEV], [libudev >= 216])
diff --git a/src/lib/Makefile.am b/src/lib/Makefile.am
index ba65f02..eb3df1c 100644
--- a/src/lib/Makefile.am
+++ b/src/lib/Makefile.am
@@ -34,7 +34,7 @@ BlockDev_1_0_gir_LIBS = libblockdev.la
 BlockDev_1_0_gir_INCLUDES = GObject-2.0
 BlockDev_1_0_gir_PACKAGES = devmapper libudev libcryptsetup
 BlockDev_1_0_gir_CFLAGS = -I${srcdir}/../utils/
-BlockDev_1_0_gir_LDFLAGS = -lm -ldmraid -L${builddir}/../utils/ -lbd_utils
+BlockDev_1_0_gir_LDFLAGS = -lm -L${builddir}/../utils/ -lbd_utils
 BlockDev_1_0_gir_SCANNERFLAGS = --warn-error --warn-all --identifier-prefix=BD --symbol-prefix=bd
 BlockDev_1_0_gir_EXPORT_PACKAGES = blockdev
 
diff --git a/src/utils/Makefile.am b/src/utils/Makefile.am
index d308442..055ce76 100644
--- a/src/utils/Makefile.am
+++ b/src/utils/Makefile.am
@@ -1,7 +1,7 @@
 lib_LTLIBRARIES = libbd_utils.la
 libbd_utils_la_CFLAGS = $(GLIB_CFLAGS) -Wall -Wextra -Werror
 libbd_utils_la_LDFLAGS = -version-info 1:1:1
-libbd_utils_la_LIBADD = $(GLIB_LIBS)
+libbd_utils_la_LIBADD = -lm $(GLIB_LIBS)
 libbd_utils_la_SOURCES = utils.h exec.c exec.h sizes.c sizes.h
 
 libincludedir = $(includedir)/blockdev
